RTNeural/activation/activation_eigen.h

```diff
@@ -464,6 +464,418 @@ class PReLUActivationT
+        });
+
+        std::copy(outVec.data(), outVec.data() + Layer<T>::in_size, out);
+    }
+
+    Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> inVec;
+    Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> outVec;
+};
+
+/** Static implementation of an approximated GeLU activation layer. */
+template <typename T, int size, typename MathsProvider = DefaultMathsProvider>
+class GeLUActivationT
+{
+    using v_type = Eigen::Matrix<T, size, 1>;
+
+public:
+    static constexpr auto in_size = size;
+    static constexpr auto out_size = size;
+
+    GeLUActivationT()
+        : outs(outs_internal)
+    {
+        outs = v_type::Zero();
+    }
+
+    std::string getName() const noexcept { return "gelu"; }
+
+    constexpr bool isActivation() const noexcept { return true; }
+
+    RTNEURAL_REALTIME void reset() { }
+
+    RTNEURAL_REALTIME inline void forward(const v_type& ins) noexcept
+    {
+        static const T sqrt_2_over_pi = T(0.45015815807);
+        outs = ins.array().unaryExpr([sqrt_2_over_pi](T x) {
+            T xCube = x * x * x;
+            return 0.5 * x * (1.0 + MathsProvider::tanh(sqrt_2_over_pi * (x + 0.044715 * xCube)));
+        });
+    }
+
+    Eigen::Map<v_type, RTNeuralEigenAlignment> outs;
+
+private:
+    T outs_internal alignas(RTNEURAL_DEFAULT_ALIGNMENT)[out_size];
+};
+
+/** Dynamic implementation of a Swish activation layer. */
+template <typename T, typename MathsProvider = DefaultMathsProvider>
+class SwishActivation : public Activation<T>
+{
+public:
+    /** set optional beta parameter (default is 1) */ 
+    explicit SwishActivation(int size, T beta = (T)1)
+        : Activation<T>(size, {}, "swish"), beta(beta)
+    {
+        inVec = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>::Zero(size, 1);
+        outVec = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>::Zero(size, 1);
+    }
+
+    SwishActivation(std::initializer_list<int> sizes, T beta = (T)1)
+        : SwishActivation(*sizes.begin(), beta)
+    {
+    }
+
+    RTNEURAL_REALTIME inline void forward(const T* input, T* out) noexcept override
+    {
+        inVec = Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic, 1>, RTNeuralEigenAlignment>(
+            input, Layer<T>::in_size, 1);
+
+        /** apply swish: x * sigmoid(beta * x) */
+        outVec = inVec.array() * MathsProvider::sigmoid(beta * inVec.array());
+        std::copy(outVec.data(), outVec.data() + Layer<T>::in_size, out);
+    }
+
+    Eigen::Matrix<T, Eigen::Dynamic, 1> inVec;
+    Eigen::Matrix<T, Eigen::Dynamic, 1> outVec;
+
+private:
+    T beta;
+};
+
+/** Static implementation of a Swish activation layer. */
+template <typename T, int size, typename MathsProvider = DefaultMathsProvider>
+class SwishActivationT
+{
+    using v_type = Eigen::Matrix<T, size, 1>;
+
+public:
+    static constexpr auto in_size = size;
+    static constexpr auto out_size = size;
+
+    /** set optional beta parameter (default is 1) */ 
+    SwishActivationT(T beta = (T)1)
```

Comment:
It would be nice if `beta` could be determined at compile-time.... Maybe have a look at how the `ELuActivationT` layer does something similar? (https://github.com/jatinchowdhury18/RTNeural/blob/main/RTNeural/activation/activation_eigen.h#L327)

Reply:
Sure I’ll rework it to be in the same vein as ELU